src/Pages/Shared/CommandBar/CommandBarBackdrop.tsx

```diff
@@ -0,0 +1,346 @@
+import { useEffect, useMemo, useRef, useState } from 'react'
+import { useHistory } from 'react-router-dom'
+
+import {
+    API_STATUS_CODES,
+    Backdrop,
+    GenericFilterEmptyState,
+    getUserPreferences,
+    KeyboardShortcut,
+    logExceptionToSentry,
+    ResponseType,
+    SearchBar,
+    stopPropagation,
+    SupportedKeyboardKeysType,
+    ToastManager,
+    ToastVariantType,
+    updateUserPreferences,
+    useQuery,
+    useRegisterShortcut,
+    UserPreferencesType,
+} from '@devtron-labs/devtron-fe-common-lib'
+
+import CommandGroup from './CommandGroup'
+import { NAVIGATION_GROUPS, RECENT_ACTIONS_GROUP, RECENT_NAVIGATION_ITEM_ID_PREFIX, SHORT_CUTS } from './constants'
+import { CommandBarBackdropProps, CommandBarGroupType } from './types'
+import { getNewSelectedIndex, sanitizeItemId } from './utils'
+
+const CommandBarBackdrop = ({ handleClose }: CommandBarBackdropProps) => {
+    const history = useHistory()
+    const { registerShortcut, unregisterShortcut } = useRegisterShortcut()
+
+    const [searchText, setSearchText] = useState('')
+    const [selectedItemIndex, setSelectedItemIndex] = useState(0)
+
+    const { data: recentActionsGroup, isLoading } = useQuery({
+        queryFn: ({ signal }) =>
+            getUserPreferences(signal).then((response) => {
+                const responseData: ResponseType<typeof response> = {
+                    code: API_STATUS_CODES.OK,
+                    status: 'OK',
+                    result: response,
+                }
+                return responseData
+            }),
+        queryKey: ['recentNavigationActions'],
+        select: ({ result }) =>
+            result.commandBar.recentNavigationActions.reduce<CommandBarGroupType>((acc, action) => {
+                const requiredGroup = structuredClone(NAVIGATION_GROUPS).find((group) =>
```

Comment:
```suggestion
                const requiredGroup = NAVIGATION_GROUPS.find((group) =>
```

structuredClone is not required

Reply:
Mutating requiredGroup id which will mess up NAVIGATION_GROUPS const